From 207e8bf4bdb8dbdf8800d4b5cbac5af746047f37 Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@arm.com>
Date: Fri, 14 May 2021 14:06:31 +0100
Subject: [PATCH] generate_binary: add genpeimg option

Add a --genpeimg option as the meson.build uses find_program(), so the
binary could be anywhere.
---
 efi/generate_binary.py | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/efi/generate_binary.py b/efi/generate_binary.py
index 0193e50..907af9f 100755
--- a/efi/generate_binary.py
+++ b/efi/generate_binary.py
@@ -48,13 +48,15 @@ def _run_objcopy(args):
 
 
 def _run_genpeimg(args):
+    if not args.genpeimg:
+        return
 
-    # this is okay if it does not exist
-    argv = ["genpeimg", "-d", "+d", "+n", "-d", "+s", args.outfile]
+    argv = [args.genpeimg, "-d", "+d", "+n", "-d", "+s", args.outfile]
     try:
         subprocess.run(argv, check=True)
     except FileNotFoundError as _:
-        pass
+        print(str(e))
+        sys.exit(1)
 
 
 if __name__ == "__main__":
@@ -63,6 +65,7 @@ def _run_genpeimg(args):
     parser.add_argument(
         "--objcopy", default="objcopy", help="Binary file to use for objcopy"
     )
+    parser.add_argument("--genpeimg", help="Binary file to use for genpeimg")
     parser.add_argument("--arch", default="x86_64", help="EFI architecture")
     parser.add_argument("infile", help="Input file")
     parser.add_argument("outfile", help="Output file")
